photos/user.go

```diff
@@ -0,0 +1,216 @@
+	{deletePhotoOp, "delete photo", 1, 0},
+	{deleteCommentOp, "delete comment", 2.5, 0},
+}
+
+var mu sync.Mutex
+var computing = false
+var totalOps int
+var maxUserID int
+var noUserOps int
+var failedOps int
+var opCounts = map[int]int{}
+var histogram = map[int]int{}
+
+func init() {
+	// Compute the total of all op relative frequencies.
+	var relFreqTotal float64
+	for _, op := range ops {
+		relFreqTotal += op.relFreq
+	}
+	// Normalize frequencies.
+	var normFreqTotal float64
+	for _, op := range ops {
+		normFreq := op.relFreq / relFreqTotal
+		op.normFreq = normFreqTotal + normFreq
+		normFreqTotal += normFreq
+	}
+}
+
+// randomOp chooses a random operation from the ops slice.
+func randomOp() *opDesc {
+	r := rand.Float64()
+	for _, op := range ops {
+		if r < op.normFreq {
+			return op
+		}
+	}
+	return ops[len(ops)-1]
+}
+
+// startUser simulates a stream of user events until the stopper
+// indicates it's time to exit.
+func startUser(ctx Context, stopper *stop.Stopper) {
+	var lastOps int
+	ticker := time.NewTicker(statsInterval)
+	for {
+		select {
+		case <-stopper.ShouldStop():
+			mu.Lock()
+			if computing {
+				return
+			}
+			computing = true
+			mu.Unlock()
+			computeHistogram()
+			return
+		case <-ticker.C:
+			mu.Lock()
+			opsPerSec := float64(totalOps-lastOps) / float64(statsInterval/1E9)
+			log.Infof("%d ops, %d no-user, %d errs (%.2f/s) across %d users",
+				totalOps, noUserOps, failedOps, opsPerSec, len(histogram))
+			lastOps = totalOps
+			mu.Unlock()
+		default:
+			userID := 1 + int(rand.ExpFloat64()/rate)
+			if userID > maxUserID {
+				maxUserID = userID
+			}
+			op := randomOp()
+
+			stopper.RunTask(func() {
+				err := runUserOp(ctx, userID, op.typ)
+				mu.Lock()
+				totalOps++
+				histogram[userID]++
+				opCounts[op.typ]++
+				switch {
+				case err == noUserErr:
+					noUserOps++
+				case err != nil:
+					failedOps++
+					log.Errorf("failed to run %s op for %d: %s", op.name, userID, err)
+				}
+				mu.Unlock()
+			})
+		}
+	}
+}
+
+// runUserOp starts a transaction and creates the user if it doesn't
+// yet exist.
+func runUserOp(ctx Context, userID, opType int) error {
+	return executeTxn(ctx.DB, func(tx *sql.Tx) error {
+		switch opType {
+		case createUserOp:
+			return createUser(tx, userID)
+		case createPhotoOp:
+			return createPhoto(tx, userID)
+		case createCommentOp:
+			return createComment(tx, userID)
+		case listPhotosOp:
+			return listPhotos(tx, userID, nil)
+		case listCommentsOp:
+			_, err := listComments(tx, userID, nil)
+			return err
+		case updatePhotoOp:
+			return updatePhoto(tx, userID)
+		case updateCommentOp:
+			return updateComment(tx, userID)
+		case deletePhotoOp:
+			return deletePhoto(tx, userID)
+		case deleteCommentOp:
+			return deleteComment(tx, userID)
+		default:
+			return util.Errorf("unsupported op type: %d", opType)
+		}
+		return nil
+	})
+}
+
+type bucket struct {
+	min, max, count, total int
+}
+
+func computeHistogram() {
```

Comment:
Sure...this just grew organically as I was verifying the distribution of user events.